tests/controls/gridtest.cpp

```diff
@@ -31,32 +31,165 @@
 namespace
 {
 
-// Derive a new class inheriting from wxGrid just to get access to its
+wxString CellCoordsToString(int row, int col)
+{
+    return wxString::Format("(%d, %c)", 1 + row, 'A' + col);
+}
+
+wxString CellSizeToString(int rows, int cols)
+{
+    return wxString::Format("%dx%d", rows, cols);
+}
+
+struct Multicell
+{
+    int row, col, rows, cols;
+
+    Multicell(int row, int col, int rows, int cols)
+        : row(row), col(col), rows(rows), cols(cols) { }
+
+    wxString Coords() const { return CellCoordsToString(row, col); }
+
+    wxString Size() const { return CellSizeToString(rows, cols); }
+
+    wxString ToString() const
+    {
+        wxString s;
+
+        if ( rows == 1 && cols == 1 )
+            s = "cell";
+        else
+        {
+            s = Size() + " ";
+            if ( rows > 1 || cols > 1 )
+                s += "multicell";
+            else
+                s += "inside cell";
+        }
+
+        return wxString::Format( "%s at %s", s, Coords() );
+    }
+};
+
+// Stores insertion/deletion info of rows or columns.
+struct EditInfo
+{
+    int pos, count, isRow;
+
+    EditInfo(int pos, int count, int isRow)
+        : pos(pos), count(count), isRow(isRow) { }
+};
+
+// Derive a new class inheriting from wxGrid, also to get access to its
 // protected GetCellAttr(). This is not pretty, but we don't have any other way
 // of testing this function.
 class TestableGrid : public wxGrid
 {
 public:
     explicit TestableGrid(wxWindow* parent)
-        : wxGrid(parent, wxID_ANY)
+        : wxGrid(parent, wxID_ANY), m_edit(0, 0, 0)
```

Comment:
Also very minor, but maybe give `EditInfo` a default ctor instead (or just add `= 0` to the existing one)?

Reply:
In this case I'll add `= 0`'s. This was a later change needed because `{ 0 }` requires C++-11.